### PR TITLE
add description kwarg in CogMeta

### LINKED ISSUE #6024: Custom description for Cogs
## The Problem

Cogs don't seem to allow their description to be overridden except for providing a docstring on the class. This becomes a problem if we want to put technical documentation on the docstring or otherwise change the description during program execution.

## The Ideal Solution

Allow the description to be set during instantiation (in the metaclass `CogMeta`), similar to how name and command_attrs is set (https://github.com/Rapptz/discord.py/blob/abe54f5bdc48b2a92fd2a0d158ac15655049174d/discord/ext/commands/cog.py#L94). 

In addition, allow description to be assignable (currently it is a read-only property) on the cog.

## The Current Solution

Set docstring (but we lose the option to configure the description later + are forced to limit what is in it)

## Summary

Allow cog description to be better customisable

### PR DESCRIPTION
**Pull request by SebbyLaw**
## Summary

Resolves #6024

Not sure if this is considered a breaking change or not.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by ShineyDev** [comment]
To resolve the linked issue this PR should also make Cog.description a writable property.

> Not sure if this is considered a breaking change or not.

Adding a new optional kwarg is not a breaking change.

File: discord/ext/commands/cog.py [hunk 1]
```diff
@@ -93,6 +95,10 @@ def __new__(cls, *args, **kwargs):
         attrs['__cog_name__'] = kwargs.pop('name', name)
         attrs['__cog_settings__'] = command_attrs = kwargs.pop('command_attrs', {})
 
+        attrs['description'] = description = kwargs.pop('description', None)
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:98]:
```suggestion
        attrs['__cog_description__'] = description = kwargs.pop('description', None)
```

File: discord/ext/commands/cog.py [hunk 2]
```diff
@@ -93,6 +95,10 @@ def __new__(cls, *args, **kwargs):
...
+        if description is not None:
+            attrs['description'] = inspect.getdoc(cls)
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:100]:
```suggestion
            attrs['__cog_description__'] = inspect.cleandoc(attrs.get('__doc__', ''))
```

File: discord/ext/commands/cog.py [hunk 3]
```diff
@@ -156,6 +162,11 @@ class Cog(metaclass=CogMeta):
 
     When inheriting from this class, the options shown in :class:`CogMeta`
     are equally valid here.
+
+    Attributes
+    -----------
+    description: :class:`str`
+        Returns the cog's description, typically the cleaned docstring.
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:169]:
```suggestion
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
I prefer having the property rather than a plain class attribute since this makes it less likely for the user to shoot themselves on the foot when they write e.g. `cog.description = "hi"` which should error out since it did before. I guess it's up to debate whether having it read-only is necessary though.

This probably also needs a versionadded.

File: discord/ext/commands/cog.py [hunk 4]
```diff
@@ -93,6 +98,10 @@ def __new__(cls, *args, **kwargs):
         attrs['__cog_name__'] = kwargs.pop('name', name)
         attrs['__cog_settings__'] = command_attrs = kwargs.pop('command_attrs', {})
 
+        attrs['__cog_description__'] = description = kwargs.pop('description', None)
+        if description is None:
+            attrs['__cog_description__'] = inspect.cleandoc(attrs.get('__doc__', ''))
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:103]:
```suggestion
        description = kwargs.pop('description', None)
        if description is None:
            description = inspect.cleandoc(attrs.get('__doc__', ''))
        attrs['__cog_description__'] = description
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
If this is meant to be modifiable then adding a property setter would probably work at this point.

**Review by Rapptz — APPROVED** [review]

**Comment by Harmon758** [comment]
This was merged with https://github.com/Rapptz/discord.py/commit/4adbe03d7ca03fce3411bea7bcd9e3619513ad94 and can be closed now.

**Pull request closed without merging**